package/MDAnalysis/analysis/dssp/dssp.py

```diff
@@ -0,0 +1,364 @@
+    - 'E' represents 'extended strand', participating in beta-ladder (parallel
+      or antiparallel)
+    - '-' represents unordered part
+
+    The implementation was taken from the pydssp package (v. 0.9.0)
+    https://github.com/ShintaroMinami/PyDSSP by Shintaro Minami.
+
+    Parameters
+    ----------
+    atoms : Union[Universe, AtomGroup]
+        input Universe or AtomGroup. In both cases, only protein residues will
+        be chosen. Heavy atoms of the protein are then selected by name
+        ("N", "CA", "C", "O O1 OT1"), and hydrogens are selected by name "H".
+    guess_hydrogens : bool, optional
+        whether you want to guess hydrogens positions, by default ``True``.
+        Guessing is made assuming perfect 120 degrees for all bonds that N
+        atom makes, and a N-H bond length of 1.01 A.
+        If ``guess_hydrogens`` is False, hydrogen atom positions on N atoms
+        will be parsed from the trajectory, except for the "hydrogen" atom
+        positions on PRO residues, and an N-terminal residue.
+
+    Raises
+    ------
+    ValueError
+        if ``guess_hydrogens`` is True but some non-PRO hydrogens are missing.
+
+    Examples
+    --------
+
+    For example, you can assign secondary structure for a single PDB file:
+
+    >>> from MDAnalysis.analysis.dssp import DSSP
+    >>> from MDAnalysisTests.datafiles import PDB
+    >>> import MDAnalysis as mda
+    >>> u = mda.Universe(PDB)
+    >>> run = DSSP(u).run()
+    >>> print("".join(run.results.dssp[0][:20]))
+    --EEEEE-----HHHHHHHH
+
+    Also, per-frame dssp assignment allows you to build average
+    secondary structure -- ``DSSP.results.dssp_ndarray`` holds
+    (n_frames, n_residues, 3) shape ndarray with one-hot encoding of
+    loop, helix and sheet, respectively:
+
+    >>> from MDAnalysis.analysis.dssp import translate, DSSP
+    >>> from MDAnalysisTests.datafiles import TPR, XTC
+    >>> u = mda.Universe(TPR, XTC)
+    >>> long_run = DSSP(u).run()
+    >>> mean_secondary_structure = translate(long_run.results.dssp_ndarray.mean(axis=0))
+    >>> print(''.join(mean_secondary_structure)[:20])
+    -EEEEEE------HHHHHHH
+
+    .. versionadded:: 2.8.0
+    """
+
+    def __init__(self, atoms: Union[Universe, AtomGroup], guess_hydrogens: bool = True):
+        self._guess_hydrogens = guess_hydrogens
+
+        ag: AtomGroup = (
+            atoms.select_atoms("protein") if isinstance(atoms, Universe) else atoms
+        )
+        super().__init__(ag.universe.trajectory)
+
+        # define necessary selections
+        # O1/OT1 is for C-terminal residue
+        heavyatom_names = ("N", "CA", "C", "O O1 OT1")
+        self._heavy_atoms: dict[str, "AtomGroup"] = {
+            t: ag.select_atoms(f"protein and name {t}") for t in heavyatom_names
+        }
+        self._hydrogens: list["AtomGroup"] = [
+            res.atoms.select_atoms("name H")
+            for res in ag.select_atoms("protein").residues
```

Comment:
I couldn't make the H-named atoms quickly, since I can't figure out how to account for missing H's on prolines, and I really need them later. I didn't spend much time on that, but I could open an enhancement issue on github after it gets merged.